src/mount.c

```diff
@@ -874,3 +880,62 @@ cc_get_device_and_fstype(gchar *mount_point, gchar **device_name, gchar **fstype
 	endmntent(file);
 	return ret;
 }
+
+/*!
+ * Check for devicemapper storage for container rootfs.
+ * If devicemapper block device is detected, set the
+ * config->device_name to the device found.
+ *
+ * \param config \ref cc_oci_config.
+ *
+ * \return \c true on success, else \c false.
+ */
+gboolean
+cc_oci_check_for_devicemapper(struct cc_oci_config *config)
+{
+       uint major, minor;
+       char *fstype = NULL;
+
+       if (! config) {
+               return false;
+       }
+
+       gchar *container_root = config->oci.root.path;
+
+       if (! cc_device_for_path(container_root, &major, &minor)) {
+               g_critical("Could not get the underlying device for rootfs");
+               return false;
+       }
+
+       if (! cc_is_devicemapper(major, minor)) {
```

Comment:
Could we make that one a little more generic, and have a `cc_is_blockdevice()` that could eventually also support other block device based graph drivers ?